.github/CONTRIBUTING.md

```diff
@@ -547,6 +547,91 @@ for (var/month in 1 to 12)
 for (var/i in reagents)
 ```
 
+### Icons are for image manipulation and defining an obj's `.icon` var, appearances are for everything else.
+BYOND will allow you to use a raw icon file or even an icon datum for underlays, overlays, and what not (you can even use strings to refer to an icon state on the current icon). The issue is these get converted by BYOND to appearances on every overlay insert or removal involving them, and this process requires inserting the new appearance into the global list of appearances, and informing clients about them.
+
+Converting them yourself to appearances and storing this converted value will ensure this process only has to happen once for the lifetime of the round. Helper functions exist to do most of the work for you.
+
+
+Bad:
+```dm
+/obj/machine/update_overlays(var/blah)
```

Comment:
```suggestion
/obj/machine/update_overlays(blah)
```
No need for `var` boilerplate in the arguments. They can't be anything besides variables.
Same for the other entries.